libmcount/dynamic.c

```diff
@@ -67,6 +64,30 @@ static struct mcount_orig_insn *lookup_code(struct rb_root *root,
 	return iter;
 }
 
+static struct mcount_orig_insn *lookup_code(struct rb_root *root,
+					    unsigned long addr)
+{
+	struct rb_node *parent = NULL;
+	struct rb_node **p = &root->rb_node;
+	struct mcount_orig_insn *iter;
+
+	while (*p) {
+		parent = *p;
+		iter = rb_entry(parent, struct mcount_orig_insn, node);
+
+		if (iter->addr == addr)
+			return iter;
+
+		if (iter->addr > addr)
+			p = &parent->rb_left;
+		else
+			p = &parent->rb_right;
+	}
+
+	return NULL;
+}
+
```

Comment:
Why?  If you do because it needs to add `INSTRUMENT_SIZE` then I think it's better doing it before calling this.  Please see my comments in the next commit.